### PR TITLE
[Enhance] enhance logging (mostly on token stats)

### PR DESCRIPTION
**Pull request by nil0x9**
1. use reduced tokens stats for exp_tgs;
2. rename some variables (drop the reduced_ prefix, use total_ instead for clarity);
3. drop maxvio stats in loss_log as it is already covered in internal metrics.

### CODE REVIEW

File: xtuner/v1/train/trainer.py [hunk 1]
```diff
@@ -762,26 +762,24 @@ def fit(self):
...
 
             internal_metrics = self._maybe_pop_model_internal_metrics(engine_input)
 
             self._cur_step += 1
 
             reduced_step_consumed_tokens = self._reduce_number_across_rank(step_consumed_tokens)
-            self._reduced_consumed_tokens += reduced_step_consumed_tokens
-
-            self._exp_consumed_tokens += step_consumed_tokens
+            self._total_consumed_tokens += reduced_step_consumed_tokens
```
Comment (jayhenry) [inline xtuner/v1/train/trainer.py]: 这里命名要注意保持和loss在代码中和日志中的一致。
比如 total_loss表示单卡上总loss（llm, balance, zloss）， reduced_llm_loss表示各卡聚合后llm_loss。
所以建议 要么保留原来的 reduced_consumed_tokens命名，要么将 loss的命名也改成和这里一致。

Comment (nil0x9) [reply xtuner/v1/train/trainer.py]: 这里确实有歧义，在现在的 codebase 里，`total_` 被用来表示时间维度上的总和 (e.g., `total_[step|epoch|iter]`），也有被用来表达空间上的总和（`total_GPU_per_node`），`total_loss`是表示几种 loss 的聚合， 显得有点混乱。感觉是应该有个好的 naming convention 来把这些统一的区分开。

这里的`_reduced_consumed_tokens` 其实同时兼有 时间维度上的累积，和空间维度的聚合两个含义，所以感觉有点奇怪，我想一想要怎么改善这个 PR。

Comment (nil0x9) [reply xtuner/v1/train/trainer.py]: > 这里命名要注意保持和loss在代码中和日志中的一致。 比如 total_loss表示单卡上总loss（llm, balance, zloss）， reduced_llm_loss表示各卡聚合后llm_loss。 所以建议 要么保留原来的 reduced_consumed_tokens命名，要么将 loss的命名也改成和这里一致。

我现在把 `total_loss` 修改为 `local_loss`，同时还附带把原来 `extra_log_info` 里的一个叫 `loss` 的 key 重新命名（这个值对 pt/sft 而言是 reduce_sum 之前的 ce_loss，在 tensorboard 上很容易引发歧义）.

我理想来讲，step vs. 整个训练周期、单 rank vs. 所有 rank、 单个值 vs. 总和，都要有个前缀做区分，但是这样每个变量都会显得非常冗余，个人浅见只能退而求其次，做一些合理的省略（只要上下文是清晰的）：

1. 对于 loss 而言，几个 loss 加和，理论上只要叫 loss 即可，细分的 loss 用 xx_loss 做区别；
2. 对于 loss 或者 tokens 而言，需要区分单 rank 和全局的求和或平均，由于单 rank 的 log 不多，如无特别前缀，都表示 reduced 之后的 版本，rank 上的值用 `local_` 前缀表示；
3.  时间维度上仍然需要区分单步统计值或者累计值，主要针对 tokens 而言，loss 没有这类需求，step 表示当前步， total 表示整个训练周期。

File: xtuner/v1/engine/train_engine.py [hunk 2]
```diff
@@ -346,7 +346,7 @@ def train_step(self, data_batches: list[ModelItem]) -> tuple[LossLog, OtherLog]:
             reduced_z_loss = step_z_loss
             dist.all_reduce(reduced_z_loss.div_(dist.get_world_size()))
             loss_log["reduced_z_loss"] = reduced_z_loss.item()
-        other_log["consumed_tokens"] = step_consumed_tokens.item()
+        other_log["consumed_tokens"] = cast(int, step_consumed_tokens.item())
```
Comment (jayhenry) [inline xtuner/v1/engine/train_engine.py]: rename to ` other_log["step_consumed_tokens"]` to keep consistent

File: xtuner/v1/engine/vision_compose_train_engine.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def train_step(self, data_batches: List[ModelItem]) -> tuple[LossLog, OtherLog]:
             reduced_z_loss = step_z_loss
             dist.all_reduce(reduced_z_loss.div_(dist.get_world_size()))
             loss_log["reduced_z_loss"] = reduced_z_loss.item()
-        other_log["consumed_tokens"] = step_consumed_tokens.item()
+        other_log["consumed_tokens"] = cast(int, step_consumed_tokens.item())
```
Comment (jayhenry) [inline xtuner/v1/engine/vision_compose_train_engine.py]: rename to  other_log["step_consumed_tokens"]

File: xtuner/v1/model/utils/misc.py
Comment (jayhenry) [inline xtuner/v1/model/utils/misc.py]: rename `log_rank_loss` to `local_base_loss`

File: xtuner/v1/train/trainer.py (hunk 1 above)
Comment (jayhenry) [reply xtuner/v1/train/trainer.py]: Good! 总结下前缀规则就是：
空间上，rank的用 local_，默认reduced无前缀。
时间上，用step_和total_。

File: xtuner/v1/model/utils/misc.py
Comment (nil0x9) [reply xtuner/v1/model/utils/misc.py]: OK
